octopoes/octopoes/models/__init__.py

```diff
@@ -11,8 +11,8 @@
 class Reference(str):
     @classmethod
     def parse(cls, ref_str: str) -> tuple[str, str]:
-        object_type, *natural_key_parts = ref_str.split("|")
-        return object_type, "|".join(natural_key_parts)
+        object_type, *natural_key_parts = ref_str.split("|", 1)
+        return object_type, natural_key_parts[0] if natural_key_parts else ""
```

Comment:
This isn't functionally the same and also not really merging different loops, is this supposed to be in this PR? If so I think we need to look at this more carefully as this seems to change reference objects everywhere and would result in objects that have multiple natural keys parts to only return the first part as natural key?

The Reference unit tests seem to only test objects that have a natural key with a single part.

Reply:
Sorry, no, it was an optimization that I spotted and chose to include in this PR, its probably better of in a separate Pr.
Functionally, this does not change the outcome whether or not multiple pipes are present in the natural_key.

Reply:
I don't think we really have to split simple optimizations like this, but I think it would be good to mention in the PR title and description that there are multiple optimizations in a PR next time.